libtsuba/src/ParquetReader.cpp

```diff
@@ -322,7 +316,13 @@ class BlockedParquetReader {
       }
       tables.emplace_back(arrow::Table::Make(arrow::schema(fields), columns));
     }
-    return KATANA_CHECKED(arrow::ConcatenateTables(tables));
+
+    auto concatenated_tables = arrow::ConcatenateTables(tables);
```

Comment:
This should be something like :
```
std::shared_ptr<arrow::Table> concatenated_table = KATANA_CHECKED(arrow::ConcatenateTables(tables));
```
Then just drop the `ValueOrDie`'s below. KATANA_CHECKED will check for an arrow error, translate it into one of our errors and return it if it finds one.